### DISCUSSION TITLE
How to differentiate peers in Typescript

### DISCUSSION THREAD

**Discussion by jackadair** [tailwindlabs/tailwindcss#10304] [Help]
As per the documentation I should be able to differentiate peers like so:

```
<fieldset>
  <legend>Published status</legend>

  <input id="draft" class="peer/draft" type="radio" name="status" checked />
  <label for="draft" class="peer-checked/draft:text-sky-500">Draft</label>

  <input id="published" class="peer/published" type="radio" name="status" />
  <label for="published" class="peer-checked/published:text-sky-500">Published</label>

  <div class="hidden peer-checked/draft:block">Drafts are only visible to administrators.</div>
  <div class="hidden peer-checked/published:block">Your post will be publicly visible on your site.</div>
</fieldset>
```

However when I attempt to use this in my own project it does not work. Is there some TypeScript-specific syntax that I am missing to get this to work?

```
         <input
          className="hidden peer/buying" type="radio" name={"page1"}value={"buying"} id={"buying"}
        />
        <label
          htmlFor="buying"
          className=" peer-checked/buying:bg-primary-25 peer-checked/buying:border-primary w-1/3 h-14 mb-2 border-2 border-base-25 bg-base-5 mx-auto font-sofia rounded-xl "
        >
          Buying
        </label>
        <input className="hidden peer/selling" type="radio" name={"page1"} value={"selling"} id={"selling"}/>
        <label
          htmlFor="selling"
          className="peer-checked/selling:bg-primary-25 peer-checked/selling:border-primary w-1/3 h-14 mb-2 border-2 border-base-25  bg-base-5 mx-auto font-sofia rounded-xl "
        >
          Selling
        </label>
```

Thanks for the help.

**Comment by wongjn** [comment]
Seems to be working in [Tailwind Play](https://play.tailwindcss.com/EoVmejQQZ7) which suggests there could be some misconfiguration within your project. What behavior "does not work"? How do you expect it to work vs its current behavior? Consider checking:
- the `content` file globs sufficiently cover the file that this code resides in.
- string interpolation/concatenation is not used to construct the class name string.

**Reply by jackadair** [reply]
The behaviour that is not working in **TypeScript** is as follows:

> Differentiating peers
When using multiple peers, you can style something on the state of a specific peer by giving that peer a unique name using a peer/{name} class, and including that name in modifiers using classes like peer-checked/{name}

I am able to achieve the desired behaviour when using peer-based styling without differentiators such as this example from the documentation:

```<form>
  <label className="block">
    <span className="block text-sm font-medium text-slate-700">Email</span>
    <input type="email" className="peer ..."/>
    <p className="mt-2 invisible peer-invalid:visible text-pink-600 text-sm">
      Please provide a valid email address.
    </p>
  </label>
</form>
```

Notice how this example does not use **differentiators**. ie `peer/buying`

However peer-based styling does not apply when using peer differentiators. The reason I believe this to be a TypeScript issue is because I can achieve the desired behaviour on Tailwind Play but not when I use the examples in my TypeScript Project

**Reply by wongjn** [reply]
I would not expect it to be a Typescript problem – Tailwind does not "know" the language of markup it scans. For the most part, it only sees files as strings. There could be some misconfiguration in the project and when abstracting to Tailwind Play, the misconfiguration is not ported across. Or perhaps there is some other part of the page that is not ported into Tailwind Play when you test and thus the unexpected behavior does not show. I take a closer look if you could provide a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by jackadair** [reply]
I think misconfiguration is likely. Just confirmed with my co-worker that they had also attempted to use differentiators and was unsuccessful. I will leave this up for a bit to see if anyone else has ran into anything similar.

If I find the time I will try a fresh project. Unfortunately the repo I am working in is confidential. 

Thanks for the help!

**Reply by adamwathan** [reply]
@jackadair Are you on the latest version of Tailwind? It's a very new feature.

**Reply by jackadair** [reply]
Ahh that would make sense 🤦. I am not.

Thanks for the tip!
